### PR TITLE
Bump CKMS insertion performance

### PR DESCRIPTION
**Pull request by blt**
There's a ton of work in this PR but the main show is a bump in CKMS insertion performance. A lot of intermediary work happened but ended up being dead-ends. The performance improvement is modest.

### CODE REVIEW

File: src/ckms.rs [hunk 1]
```diff
@@ -10,25 +10,48 @@
...
 #[cfg_attr(feature = "serde_support", derive(Serialize, Deserialize))]
-struct Entry<T: Copy> {
+struct Entry<T>
+where
+    T: Copy + PartialEq,
+{
     v: T,
     g: usize,
     delta: usize,
 }
```
Comment (pulltab) [inline src/ckms.rs]: Can we get a comment as to why we implement PartialEq / ParitalOrd ourselves?

File: src/ckms.rs [hunk 2]
```diff
@@ -67,7 +90,9 @@ where
         + Debug
         + std::convert::Into<f64>,
 {
-    fn add_assign(&mut self, rhs: CKMS<T>) {
+    fn add_assign(&mut self, mut rhs: CKMS<T>) {
+        rhs.flush();
+        self.flush();
```
Comment (pulltab) [inline src/ckms.rs]: What purpose does this serve?

Comment (blt) [reply src/ckms.rs]: Dead code. Oops!

File: src/ckms.rs (hunk 1 above)
Comment (blt) [reply src/ckms.rs]: Resolved.

**Pull request merged**